### PR TITLE
 [12.0] event_sale_registration_multi_qty: Migration to 12.0

### PR DESCRIPTION
**Pull request by chienandalu**
Depends on:

- [x] https://github.com/OCA/event/pull/138 `event_registration_multi_qty`

This module extends Event Registration Multi Qty so it can be used on
event sales.

cc @Tecnativa

### CODE REVIEW

File: event_sale_registration_multi_qty/models/sale_order.py [hunk 1]
```diff
@@ -11,15 +10,17 @@ class SaleOrderLine(models.Model):
...
         """ Writes registrations """
         Registration = self.env['event.registration']
         registrations = Registration.search([
             ('sale_order_line_id', 'in', self.ids)
         ])
         for so_line in self.filtered('event_id'):
             if not so_line.event_id.registration_multi_qty:
-                super(SaleOrderLine, so_line)._update_registrations()
+                super(SaleOrderLine, so_line)._update_registrations(
+                    confirm, cancel_to_draft, registration_data)
```
Comment (pedrobaeza) [inline event_sale_registration_multi_qty/models/sale_order.py]: Please respect keyword arguments putting them as so in the call (`confirm=confirm, cancel_to_draft=...`).

File: event_sale_registration_multi_qty/models/sale_order.py [hunk 2]
```diff
@@ -11,15 +10,17 @@ class SaleOrderLine(models.Model):
...
                 continue
             existing_registrations = registrations.filtered(
                 lambda self: self.sale_order_line_id.id == so_line.id)
```
Comment (pedrobaeza) [inline event_sale_registration_multi_qty/models/sale_order.py]: Can we pass the quantity by context instead and inject the value in the create method instead overwriting all of this code? If needed, we can change before calling super the line qty to 1 and after saving the real qty, and restoring it later.

**Review by pedrobaeza — COMMENTED** [review]
A doubt about the used method.

File: event_sale_registration_multi_qty/models/sale_order.py (hunk 2 above)
Comment (chienandalu) [reply event_sale_registration_multi_qty/models/sale_order.py]: Done

File: event_sale_registration_multi_qty/models/sale_order.py [hunk 3]
```diff
@@ -11,32 +10,26 @@ class SaleOrderLine(models.Model):
...
+            so_line.product_uom_qty = 1
+            super(SaleOrderLine, so_line)._update_registrations(
+                confirm=confirm, cancel_to_draft=cancel_to_draft,
+                registration_data=registration_data)
+            # Set the so line qty back and set the registration qty
+            so_line.product_uom_qty = product_uom_qty
+            registration = Registration.search([
+                ('sale_order_line_id', '=', so_line.id)
+            ])
+            registration.qty = int(product_uom_qty)
```
Comment (pedrobaeza) [inline event_sale_registration_multi_qty/models/sale_order.py:35]: Isn't this duplicated with https://github.com/OCA/event/pull/139/commits/cb46d203fe4872ba9eb12c286f739d4e437ebb20#diff-191ffd8beda0d9fbfe9d0541696ca4c8R15?

Comment (chienandalu) [reply event_sale_registration_multi_qty/models/sale_order.py:35]: No, because this method updates an existing attendee since the other creates it

Comment (pedrobaeza) [reply event_sale_registration_multi_qty/models/sale_order.py:35]: Can't this be unified then?

Comment (chienandalu) [reply event_sale_registration_multi_qty/models/sale_order.py:35]: On a second thought, the line in the _prepare it's not necessary due to the last refactor of this :) Sending patch

**Comment by chienandalu** [comment]
Ok, changes done

File: event_sale_registration_multi_qty/__manifest__.py [hunk 4]
```diff
@@ -1,15 +1,13 @@
-# -*- coding: utf-8 -*-
-# Copyright 2017 David Vidal <david.vidal@tecnativa.com>
-# Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
+# Copyright 2017 Tecnativa - David Vidal
```
Comment (victor-ditay) [inline event_sale_registration_multi_qty/__manifest__.py]: Here and the next line should cover year 2017-2019. @chienandalu apologies if asking again but maybe GitHub has some issues to keep my comments

File: event_sale_registration_multi_qty/models/sale_order.py [hunk 5]
```diff
@@ -1,6 +1,5 @@
-# -*- coding: utf-8 -*-
-# Copyright 2017 David Vidal <david.vidal@tecnativa.com>
-# Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
+# Copyright 2017 Tecnativa - David Vidal
```
Comment (victor-ditay) [inline event_sale_registration_multi_qty/models/sale_order.py]: Here and the next line should cover year 2017-2019. @chienandalu apologies if asking again but maybe GitHub has some issues to keep my comments

File: event_sale_registration_multi_qty/wizards/event_edit_registration.xml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!-- Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
+<!-- Copyright 2017 Tecnativa - Sergio Teruel
```
Comment (victor-ditay) [inline event_sale_registration_multi_qty/wizards/event_edit_registration.xml:2]: Here should cover year 2017-2019. @chienandalu apologies if asking again but maybe GitHub has some issues to keep my comments

File: event_sale_registration_multi_qty/wizards/event_edit_registration.py [hunk 7]
```diff
@@ -1,8 +1,6 @@
-# -*- coding: utf-8 -*-
-# Copyright 2017 David Vidal <david.vidal@tecnativa.com>
-# Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
+# Copyright 2017 Tecnativa - David Vidal
```
Comment (victor-ditay) [inline event_sale_registration_multi_qty/wizards/event_edit_registration.py]: Here and the next line should cover year 2017-2019. @chienandalu apologies if asking again but maybe GitHub has some issues to keep my comments

**Review by victor-ditay — CHANGES_REQUESTED** [review]

**Comment by chienandalu** [comment]
Thanks @victormmtorres . Changes done

**Comment by chienandalu** [comment]
Final review , please :)

**Review by victor-ditay — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**